### ISSUE TITLE
Verify that standard Spark operator works with Cloudflow.

### ISSUE THREAD

**Issue by RayRoestenburg** [lightbend/cloudflow#714]
It would be better if Cloudflow works with the standard Spark operator, for Spark 2.4.5.
We have back-ported a fix for service account for the executor, we need to check again why this is important.
If it is only important for certain scenarios we can detail these in the documentation, so that users can simply use the standard build of the Spark Operator: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/releases/tag/v1beta2-1.1.2-2.4.5

**Comment by RayRoestenburg** [comment]
(It might make sense to try and spike to get Spark 3.0 to work since this has the fix we backported to Spark 2.4.5.)

**Comment by blublinsky** [comment]
It looks like an attempt to use a standard Spark operator will require some changes in the current implementation. Here is a CR definition for Google Spark operator - https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/api-docs.md#sparkapplication, which is similar, but not the same as the one currently used. An example of CR used is below:

apiVersion: sparkoperator.k8s.io/v1beta2
kind: SparkApplication
metadata: 
  name: spark-sensors-ingress
  namespace: spark-sensors
spec: 
  driver: 
    configMaps: 
      - 
        name: configmap-spark-sensors-ingress
        path: /etc/cloudflow-runner
    javaOptions: ""
    labels: 
      app.kubernetes.io/component: streamlet
      app.kubernetes.io/managed-by: cloudflow
      app.kubernetes.io/name: spark-sensors-ingress
      app.kubernetes.io/part-of: spark-sensors
      app.kubernetes.io/version: 7-7e2843c-dirty
      com.lightbend.cloudflow/app-id: spark-sensors
      com.lightbend.cloudflow/streamlet-name: ingress
      version: "2.4.5"
    secrets: 
      - 
        name: ingress
        path: /etc/cloudflow-runner-secret
        secretType: Generic
    serviceAccount: cloudflow-app-serviceaccount
    volumeMounts: 
      - 
        mountPath: /mnt/spark/storage
        name: persistent-storage
      - 
        mountPath: /mnt/downward-api-volume/
        name: downward-api-volume
  executor: 
    configMaps: 
      - 
        name: configmap-spark-sensors-ingress
        path: /etc/cloudflow-runner
    instances: 1
    javaOptions: ""
    labels: 
      app.kubernetes.io/component: streamlet
      app.kubernetes.io/managed-by: cloudflow
      app.kubernetes.io/name: spark-sensors-ingress
      app.kubernetes.io/part-of: spark-sensors
      app.kubernetes.io/version: 7-7e2843c-dirty
      com.lightbend.cloudflow/app-id: spark-sensors
      com.lightbend.cloudflow/streamlet-name: ingress
      version: "2.4.5"
    secrets: 
      - 
        name: ingress
        path: /etc/cloudflow-runner-secret
        secretType: Generic
    volumeMounts: 
      - 
        mountPath: /mnt/spark/storage
        name: persistent-storage
      - 
        mountPath: /mnt/downward-api-volume/
        name: downward-api-volume
  image: "docker.io/lightbend/spark-sensors@sha256:d3cd2fbd0d4583f70b905d6d1a382453d784507570fdba61c5266ba7e1ca052b"
  imagePullPolicy: Always
  mainApplicationFile: spark-internal
  mainClass: cloudflow.runner.Runner
  mode: cluster
  monitoring: 
    exposeDriverMetrics: true
    exposeExecutorMetrics: true
    prometheus: 
      configFile: /etc/cloudflow-runner/prometheus.yaml
      jmxExporterJar: /prometheus/jmx_prometheus_javaagent.jar
      port: 2050
  restartPolicy: 
    onFailureRetryInterval: 10
    onSubmissionFailureRetryInterval: 60
    type: Always
  sparkConf: 
    spark.driver.cores: "1"
    spark.driver.memory: 512M
    spark.executor.cores: "1"
    spark.executor.memory: 2G
  sparkVersion: "2.4.5"
  type: Scala
  volumes: 
    - 
      name: persistent-storage
      persistentVolumeClaim: 
        claimName: spark-sensors-spark-pvc
    - 
      downwardAPI: 
        items: 
          - 
            fieldRef: 
              apiVersion: v1
              fieldPath: metadata.uid
            path: metadata.uid
          - 
            fieldRef: 
              apiVersion: v1
              fieldPath: metadata.name
            path: metadata.name
          - 
            fieldRef: 
              apiVersion: v1
              fieldPath: metadata.namespace
            path: metadata.namespace
      name: downward-api-volume

Comparing the two, it can be seen that the configuration has to be significantly reshuffled

**Comment by blublinsky** [comment]
So migration is desirable, but requires full time resource for several weeks,

**Comment by olofwalker** [comment]
>So migration is desirable, but requires full time resource for several weeks,

I think it's worth it, having our own has always caused friction.

**Comment by blublinsky** [comment]
Agreed, as a small on the side project, I captured what we generate for Spark, so that we can compare it to the CR of a "standard" Spark operator - https://docs.google.com/document/d/1Yb2w4HdFOApgd_owYsfOewKH5GGYZH_WwO6JSJg9lQY/edit

**Comment by andreaTP** [comment]
Mentioning that this PR looks active:
https://github.com/apache/spark/pull/29877/files

and will eventually backport the fixes to the `2.4.X` branch

**Comment by andreaTP** [comment]
This PR (https://github.com/apache/spark/pull/29877/files) just got merged.
As soon as there is a 2.4.8 Spark release we can test it!

**Comment by RayRoestenburg** [comment]
Awesome!

**Comment by RayRoestenburg** [comment]
Let's wait for 2.4.8 to be released and then try this again.
